### ISSUE TITLE
[MXFP8 training] Weight wrapper drops torch.nn.functional.linear bias

### ISSUE THREAD

**Issue by GotFusion** [pytorch/ao#4846]
## Description

`MXFP8TrainingWeightWrapperTensor.__torch_function__` intercepts `torch.nn.functional.linear(input, weight, bias)`, but the MXFP8 path only forwards `input` and the unwrapped `weight` to `_to_mxfp8_then_scaled_mm`. The third argument, `bias`, is ignored.

As a result, the forward computes only `input @ weight.T`, and the bias is absent from the autograd graph, leaving `bias.grad` as `None`.

The previous `MXLinear.forward` implementation in TorchAO v0.16 explicitly added `self.bias` after `mx_mm.apply`, so this appears to be a semantic regression in the weight-wrapper implementation.

The existing `test_mxfp8_training_tensor_ops_fwd_bwd` passes a bias for the linear case, but a whole-output SQNR check can miss the omission when the GEMM output dominates the bias. It also does not currently require gradients for the bias.

## Steps to reproduce

```python
import torch
import torch.nn.functional as F

from torchao.prototype.moe_training.config import (
    MXFP8TrainingOpConfig,
    MXFP8TrainingRecipe,
)
from torchao.prototype.moe_training.tensor import (
    MXFP8TrainingWeightWrapperTensor,
)

torch.manual_seed(0)
config = MXFP8TrainingOpConfig.from_recipe(
    MXFP8TrainingRecipe.MXFP8_EMULATED_RCEIL
)
x = torch.randn(32, 32, dtype=torch.bfloat16, requires_grad=True)
weight = torch.randn(32, 32, dtype=torch.bfloat16, requires_grad=True)
bias = torch.randn(32, dtype=torch.bfloat16, requires_grad=True)
wrapped_weight = MXFP8TrainingWeightWrapperTensor(weight, config)

actual = F.linear(x, wrapped_weight, bias)
without_bias = F.linear(x, wrapped_weight, None)

torch.testing.assert_close(actual, without_bias + bias)
actual.sum().backward()
assert bias.grad is not None
```

On v0.18.0 and current main, the `assert_close` fails and `bias.grad` remains `None`.

## Expected behavior

The MXFP8 wrapper should preserve `F.linear` semantics by adding a non-`None` bias to the GEMM result. This should also allow autograd to compute `bias.grad`.

## Environment

- torchao: v0.18.0 and current main at `4e42c02f0`
- torch: 2.11.0
